src/daos-serialize/daos-serialize.c

```diff
@@ -0,0 +1,263 @@
+    if (numpaths < 1) {
+        MFU_LOG(MFU_LOG_ERR, "At least one pool and container is required:"
+                MFU_ERRF, MFU_ERRP(-MFU_ERR_INVAL_ARG));
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    /* create an empty file list */
+    mfu_flist flist = mfu_flist_new();
+
+    rc = daos_init();
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to initialize daos");
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    /* DAOS vars */ 
+    daos_args_t* daos_args = daos_args_new();    
+
+    int len = strlen(argpaths[0]); 
+
+    /* default to no daos prefix, but allow someone to use one */
+    bool daos_no_prefix = true;
+    if (strncmp(argpaths[0], "daos://", 7) == 0) {
+        daos_no_prefix = false; 
+    }
+
+    int tmp_rc;
+    tmp_rc = daos_parse_path(argpaths[0], len, &daos_args->src_pool_uuid,
+                         &daos_args->src_cont_uuid, daos_no_prefix);
+    if (tmp_rc != 0 || daos_args->src_pool_uuid == NULL || daos_args->src_cont_uuid == NULL) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to resolve DAOS path");
+         rc = 1;
+    }
+    
+    tmp_rc = daos_connect(rank, daos_args->src_pool_uuid, daos_args->src_cont_uuid,
+                          &daos_args->src_poh, &daos_args->src_coh,
+                          true, false);
+    if (tmp_rc != 0) {
+        rc = 1;
+    }
+
+    /* take a snapshot and walk container to get list of objects,
+     * returns epoch number of snapshot */
+    tmp_rc = mfu_flist_walk_daos(daos_args, flist);
+    if (tmp_rc != 0) {
+        rc = 1;
+    }
+
+    /* all objects are on rank 0 at this point,
+     * evenly spread them among the ranks */
+    mfu_flist newflist = mfu_flist_spread(flist);
+
+    /* get size of local list for each rank */
+    uint64_t size = mfu_flist_size(newflist);
+
+    /* serialize pool/cont to an hdf5 file */
+    uint64_t files_written = 0;
+
+    /* only create a directory if one is passed in with output_path
+     * option, otherwise use current working dir */
+    if (output_path == NULL) {
+        char cwd[FILENAME_LEN];
+        getcwd(cwd, FILENAME_LEN);
+        if (cwd == NULL) {
+            MFU_LOG(MFU_LOG_ERR, "failed to get current working directory");
+            rc = 1;
+        }
+        output_path = MFU_STRDUP(cwd);
+    } else {
+        tmp_rc = mkdir(output_path, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+        if (tmp_rc != 0 && errno != EEXIST) {
+            MFU_LOG(MFU_LOG_ERR, "failed to create output directory");
+            rc = 1;
+        }
+    }
+
+    tmp_rc = daos_cont_serialize_hdlr(rank, output_path, &files_written,
+                                      daos_args, newflist, size);
+    if (tmp_rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to serialize container (%d)", rc);
+        rc = 1;
+    }
+
+    /* destroy snapshot after copy */
+    /* TODO consider moving this into mfu_flist_copy_daos */
```

Comment:
@daltonbohning ok, I can update the daos_connect and remove the destroy snapshot after the dsync PR is merged. Since I will probably get a few merge conflicts anyway.